File: C/Program53.c
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                //
//  File name :     Program53.c                                                                                   //
//  Description :   Program to accept character from user & check if it is capital or not                         //
//  Author :        Pallavi Omprakash Malewar                                                                     //
//  Date :          27/06/2025                                                                                    //
//                                                                                                                //  
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<stdio.h>
#include<stdbool.h>


bool CheckCapital(char cValue)
{
    bool bFlag = false;

    if((cValue >= 'A') && (cValue <= 'Z'))
    {
        bFlag = true;
    }
    else
    {
        bFlag = false;
    }
    return bFlag;
}


int main()
{
    char cValue = '\0';
    bool bRet = false;

    printf("Enter Character : ");
    scanf("%c" , &cValue);

    bRet = CheckCapital(cValue);

    if(bRet == false)
    {
        printf("%c is not a Capital Letter.\n" , cValue);
    }
    else
    {
        printf("%c is Capital Letter.\n" , cValue);
    }

    return 0;
}

File: JAVA/Program53.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Program53.java                                                      //
//  Description :   Used to Calculate Percentage                                        //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          27/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Calculate
{
    int iTotalMarks = 0, iObtainedMarks = 0;

    Calculate(int iTotalMarks, int iObtainedMarks)
    {
        this.iTotalMarks = iTotalMarks;
        this.iObtainedMarks = iObtainedMarks;
    }

    public float CalculatePercentage(int iTotalMarks , int iObtainedMarks)
    {
        float fPercentageValue = 0.0f;

        fPercentageValue = ((float)this.iObtainedMarks / (float)this.iTotalMarks) * 100;

        return fPercentageValue;
    }
}

class Program53
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iTotal = 0, iObtained = 0;
        float fRet = 0.0f;

        System.out.print("Enter Total Marks : ");
        iTotal = sobj.nextInt();

        System.out.print("Enter Obtained Marks : ");
        iObtained = sobj.nextInt();

        if(iTotal <= 0)
        {
            System.out.println("Invalid Input!\nPlease Enter Valid Input.");
        }

        if(iObtained > iTotal)
        {
            System.out.println("Please Enter Valid Input!\nObtained marks cannot be negative or greater than total marks.");
        }

        Calculate cobj = new Calculate(iTotal, iObtained);

        fRet = cobj.CalculatePercentage(iTotal, iObtained);

        System.out.println("Percetage is : " + fRet);

        sobj.close();
    }
}

